beacon-chain/blockchain/service.go

```diff
@@ -279,9 +293,15 @@ func (c *ChainService) blockProcessing(done <-chan struct{}) {
 			// to take in the block and the DAG of previously processed blocks
 			// and determine all validity conditions from those two parameters.
 			isParentHashExistent := false
-			for i := 0; i < len(c.processedBlockHashesBySlot[receivedSlotNumber-1]); i++ {
+			blockhashes, err := c.chain.retrieveBlockRegistry(receivedSlotNumber - 1)
```

Comment:
I think we should rename this to be a little bit clearer. The block registry is actually a slice of block hashes for the latest unprocessed slot right before the fork-choice rule. Maybe we can rename to `retrieveUnprocessedBlockHashes` or `pendingProcessingBlockHashes`? Thoughts?